examples/kubernetes/README.md

```diff
@@ -1,18 +1,124 @@
 
 ```
-$ kubectl get pod
-POD              IP            CONTAINER(S)   IMAGE(S)      HOST                  LABELS          STATUS    CREATED      MESSAGE
-ceph-mon-5ceys   172.17.0.13                                127.0.0.1/127.0.0.1   name=frontend   Running   49 minutes
-                               ceph-mon       ceph-daemon                                         Running   49 minutes
+$ kubectl get all --namespace=ceph
+NAME                   DESIRED      CURRENT       AGE
+ceph-mds               1            1             24s
+ceph-mon-check         1            1             24s
+NAME                   CLUSTER-IP   EXTERNAL-IP   PORT(S)    AGE
+ceph-mon               None         <none>        6789/TCP   24s
+NAME                   READY        STATUS        RESTARTS   AGE
+ceph-mds-6kz0n         0/1          Pending       0          24s
+ceph-mon-check-deek9   1/1          Running       0          24s
+```
+
+# Label your storage nodes
+
+You must label your storage nodes in order to run Ceph pods on them.
+
+```
+kubectl label node <nodename> node-type=storage
+```
+
+If you want all nodes in your Kubernetes cluster to be a part of your Ceph cluster, label them all.
+
+```
+kubectl label nodes node-type=storage --all
+```
+
+Eventually all pods will be running, including a mon and osd per every labeled node.
+
 ```
+$ kubectl get pods --namespace=ceph
+NAME                   READY     STATUS    RESTARTS   AGE
+ceph-mds-6kz0n         1/1       Running   0          4m
+ceph-mon-8wxmd         1/1       Running   2          2m
+ceph-mon-c8pd0         1/1       Running   1          2m
+ceph-mon-cbno2         1/1       Running   1          2m
+ceph-mon-check-deek9   1/1       Running   0          4m
+ceph-mon-f9yvj         1/1       Running   1          2m
+ceph-osd-3zljh         1/1       Running   2          2m
+ceph-osd-d44er         1/1       Running   2          2m
+ceph-osd-ieio7         1/1       Running   2          2m
+ceph-osd-j1gyd         1/1       Running   2          2m
+```
+
+# Mounting CephFS in a pod
+
+First you must add the admin client key to your current namespace (or the namespace of your pod).
+
+```
+kubectl create secret generic ceph-client-key --from-file=./generator/ceph-client-key
+```
+
+Now, if skyDNS is set as a resolver for your host nodes:
+
+```
+kubectl create -f ceph-mount-test.yaml --namespace=ceph
+```
+
+Otherwise you must edit the file and replace `ceph-mon.ceph` with a Pod IP. It is highly reccomended that you place skyDNS as a resolver, otherwise your configuration WILL eventually stop working as Mons are rescheduled.
+
+To get skyDNS resolution working, your resolv.conf should look something like this:
+
+```
+domain <EXISTING_DOMAIN>
+search <EXISTING_DOMAIN>
+
+search svc.cluster.local
+
+nameserver 10.0.0.10
+nameserver <EXISTING_RESOLVER_IP>
+```
+
+If your pod has issues mounting, make sure mount.ceph is installed on all nodes.
+
+```
+apt-get install ceph-fs-common
```

Comment:
To avoid having to install packages (which won't work for things like CoreOS), I wonder if it would make sense (and actually work) to modify the ceph-tools example to copy `mount.ceph` to the node - https://github.com/ceph/ceph-docker/tree/master/examples/coreos/tools